src/main/java/duke/database/Database.java

```diff
@@ -0,0 +1,94 @@
+package duke.database;
+
+import duke.exception.databaseexceptions.DatabaseNotUpdatingException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.exception.databaseexceptions.DatabaseNotLoadingException;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/** Represents the database storage for Duke using a basic text file. */
+public class Database {
+
+    //ASCII 31 is known as the "Unit Seperator"
+    //Value taken from
+    //https://stackoverflow.com/questions/492090/least-used-delimiter-character-in-normal-text-ascii-128
```

Comment:
These comments really helped me understand what the following lines of codes are doing!